lean4-infoview/src/infoview/tooltips.tsx

```diff
@@ -99,19 +99,26 @@ export const WithFadingTooltip =
     }
   }
 
+  // Workaround for warning generated by react-transition-group.
+  // See https://github.com/reactjs/react-transition-group/issues/668#issuecomment-791003824.
+  const nodeRef = React.useRef(null)
+
   return (<span ref={setRef} {...props}>
     <Transition
+      nodeRef={nodeRef}
       in={isTooltipShown}
       timeout={1500}
       onEntered={() => hideTooltip()}
     >
     {state =>
-      <Tooltip
-        reference={ref}
-        style={fadeOutStyles[state]}
-      >
-        {tooltipChildren}
-      </Tooltip>}
+      <div ref={nodeRef}>
```

Comment:
It looks like this introduced a regression: the `div`s introduce linebreaks:
<img width="125" alt="Screenshot 2024-03-25 at 11 47 05 PM" src="https://github.com/leanprover/vscode-lean4/assets/13901751/152b7980-be23-42ce-be45-6976120e4ba9">

Changing the `div` to a `span` seems to fix it. However, I would advise against pulling in `react-transition-group`. The last commit was 2 years ago. This could mean it's very stable, but the presence of issues like the one you linked (and many others) suggests it's actually just abandoned. Do you think the "No definition found" popup is essential? It's nice to have for sure, but just having the action fail seems like a clear enough signal for users, and will definitely simplify maintenance.

Reply:
Sorry for wasting your time with all those messy mistakes. Not sure why I didn't notice the broken formatting when testing the change.

I do think the popup is essential to reduce confusion and I agree that avoiding this dependency is preferable. I tried to make the fade-out animation work using `onTransitionEnd` for a while before including the dependency but gave up eventually. If you know of a nice way to implement this transition without the library, please go ahead :-)

Reply:
There probably is a way, but I don't have the time currently to experiment. If you are willing to maintain this stuff (please note: the tooltip code is *the most complicated* part of the infoview), feel free to merge (but please squash the few 'fix' commits towards the end).

Reply:
I always squash by default these days. I'm willing to maintain this change! If it breaks, I'll invest the time necessary to get rid of the dependency.